### PR TITLE
[xbyak] Update to 7.06

### PR DESCRIPTION
**Pull request by shad0wshayd3**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by shad0wshayd3** [comment]
@microsoft-github-policy-service agree

**Review by FrankXie05 — APPROVED** [review]

File: ports/xbyak/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "xbyak",
-  "version": "6.73",
+  "version-string": "7.06",
```
Comment (FrankXie05) [inline ports/xbyak/vcpkg.json:3]:
```suggestion
  "version": "7.06",
```

**Review by FrankXie05 — DISMISSED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/xbyak/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/xbyak/vcpkg.json:3]: Again: `version` cannot handle leading zeros.

**Comment by shad0wshayd3** [comment]
ready for review

**Review by data-queue — APPROVED** [review]

**Pull request merged**